File: poc/cpat-core/Models/TargetTypes/Location.cs
using cpat_core.DataAccess.TargetTypes;
using cpat_core.Models.TargetTypes;
using System;
using System.Collections.Generic;

namespace cpat_core.Models
{
    /// <summary>
    /// Physical, literal place. Able to locate it on a map.
    /// </summary>
    public class Location : AbstractTarget
    {
        public string Name { get; set; }
        public string Latitude { get; set; }
        public string Longitude { get; set; }
        
        /// <summary>
        /// A list of <c>DocumentRelation</c> objects connecting one <c>Location</c> to
        /// n-number of other pieces of information.
        /// </summary>
        /// <value></value>
        public List<DocumentRelation> DocumentRelation { get; set; }

        /// <summary>
        /// Converts a <c>LocationDto</c> object to a <c>Location</c> object.
        /// </summary>
        /// <param name="data"></param>
        /// <returns></returns>
        public static Location Translate(LocationDto data)
        {
            return new Location()
            {
                Id = data.Id,
                Name = data.Name,
                Latitude = data.Latitude,
                Longitude = data.Longitude,

                DocumentRelation = data.DocumentRelation,

                DateCreated = data.DateCreated,
                UpdatedAt = data.UpdatedAt,
                //LastModifiedBy = data.LastModifiedByUserId
            };
        }

        /// <summary>
        /// Converts a collection of <c>LocationDto</c> to a collection of <c>Location</c>.
        /// </summary>
        /// <param name="data"></param>
        /// <returns></returns>
        public static IEnumerable<Location> Translate(List<LocationDto> data)
        {
            var locationList = new List<Location>();
            data.ForEach(d =>
            {
                locationList.Add(Location.Translate(d));
            });

            return locationList;
        }
    }
}

File: cpat-client/src/data/Location.js
import AbstractTarget from "./AbstractTarget";

class Location extends AbstractTarget {
    constructor(
        id,

        name,
        latitude,
        longitude,
        documentRelation,

        dateCreated,
        updatedAt,
        lastModifiedBy,
    ) {
        super(id, dateCreated, updatedAt, lastModifiedBy);

        this._name = name;
        this._latitude = latitude;
        this._longitude = longitude;
        this._documentRelation = documentRelation;
    }

    get name() {
        return this._name;
    }
    set name(value) {
        if (value) this._name = value;
    }

    get latitude() {
        return this._latitude;
    }
    set latitude(value) {
        if (value) this._latitude = value;
    }
    
    get longitude() {
        return this._longitude;
    }
    set longitude(value) {
        if (value) this._longitude = value;
    }

    get documentRelation() {
        return this._documentRelation;
    }
    set documentRelation(value) {
        if (value) this._documentRelation = value;
    }

    apiObject() {
        return {
            id: this._id,
            name: this._name,
            latitude: this._latitude,
            longitude: this._longitude,
            documentRelation: this._documentRelation,
            dateCreated: this._dateCreated,
            updatedAt: this._updatedAt,
            lastModifiedBy: this._lastModifiedBy,
        }
    }
};

export default Location;